book/src/simple-payment-and-state-verification.md

```diff
@@ -0,0 +1,158 @@
+# Simple Payment and State Verification
+
+Thin clients using Solana require a receipt for transactions that have been
+accepted by the cluster without running a full validator node.  New fullnode
+validators that desire to boot from a known checkpoint in the network require
+some proof that the checkpoint is valid.
+
+Solana receipts provide a proof that a transaction or state have been confirmed
+by the desired set of validators to a desired amount of finality.
+
+## Trusted Validators
+
+Thin clients send an RPC request to a validator to check for a recent signature
+confirmation.  The validator responds with a confirmation count for the
+signature, and clients are able to query the validator for the error code for
+the signature.  This kind of payment or state verification depends on a trusted
+validators and is not a proof.
+
+## Receipts
+
+The basic idea for receipts is creating a chain from the transaction to a slice
+of PoH ledger that proves that the desired set of validators have voted on a
+block contain that transaction to a desired degree of finality.
+
+Clients can examine this piece of the ledger and compute the locktower lockout
+rules for each of the votes.
+
+The receipts for both, state and payments, start with a merkle path from the
+value into PoH, and chain to PoH Entries containing validator votes.
+
+### Payment Merkel Path
+
+A payment receipt is a data structure that contains a merkle path from a
+transaction to the required set of validator votes.
+
+An entry hash is a merkle tree of all the transactions sorted by signature.
+
+
+```
+                          Entry hash
+               /                           \
+            hash                             hash
+     /             \                      /       \
+    /               \                    /         \
+hash(T1, status)  hash(T2, status)   hash(T3, status)  0
+
+```
+
+Transaction status is necessary for the receipt because the state receipt is
+constructed for the block.  Two transactions over the same state can appear in
+the block, and therefore, there is no way to infer from just the state whether a
+transaction that is encoded in the ledger actually succeeded in a state commit
+or failed.  It may not be necessary to encode the full code but a single
+status bit to indicate the transaction succeeded.
+
+```
+Entry {
+    poh: Hash,
```

Comment:
Can you use all the field names from the current code to make it clear that you're not proposing name changes or new fields here?